modules/javafx.graphics/src/main/java/javafx/scene/image/PixelBuffer.java

```diff
@@ -0,0 +1,194 @@
+            throw new IllegalArgumentException("PixelBuffer dimensions must be positive (w,h > 0)");
+        }
+        if (pixelFormat.getType() != PixelFormat.getByteBgraPreInstance().getType() &&
+                pixelFormat.getType() != PixelFormat.getIntArgbPreInstance().getType()) {
+            throw new IllegalArgumentException("Unsupported PixelFormat: " + pixelFormat.getType());
+        }
+        if ((pixelFormat.getType() == PixelFormat.getByteBgraPreInstance().getType()
+                && buffer.capacity() < width * height * 4) ||
+                (pixelFormat.getType() == PixelFormat.getIntArgbPreInstance().getType()
+                && buffer.capacity() < width * height)) {
+            throw new IllegalArgumentException("Insufficient memory allocated for buffer.");
+        }
+        this.buffer = buffer;
+        this.width = width;
+        this.height = height;
+        this.pixelFormat = pixelFormat;
+    }
+
+    /**
+     * Returns the {@code java.nio.Buffer} of this {@code PixelBuffer}.
+     *
+     * @return the {@code java.nio.Buffer} of this {@code PixelBuffer}.
+     */
+    public T getBuffer() {
+        return buffer;
+    }
+
+    /**
+     * Returns the width dimension of {@code PixelBuffer}.
+     *
+     * @return the width dimension of this {@code PixelBuffer}.
+     */
+    public int getWidth() {
+        return width;
+    }
+
+    /**
+     * Returns the height dimension of {@code PixelBuffer}.
+     *
+     * @return the height dimension of this {@code PixelBuffer}.
+     */
+    public int getHeight() {
+        return height;
+    }
+
+    /**
+     * Returns the {@code PixelFormat} of the pixels in the {@code java.nio.Buffer}.
+     *
+     * @return the {@code PixelFormat} of the pixels in the {@code java.nio.Buffer}.
+     */
+    public PixelFormat getPixelFormat() {
+        return pixelFormat;
+    }
+
+    /**
+     * Updates the pixel data in {@code java.nio.Buffer} using provided {@link javafx.util.Callback}.
+     * This method must be called on the JavaFX Application thread.
+     * The {@code callback} method should update the {@code java.nio.Buffer} and
+     * return a {@code Rectangle2D} which encloses the modified region, or
+     * {@code null} to indicate that entire buffer is dirty.
+     * Pseudo-code how to use this method:
+     * <pre>{@code
+     * Callback<PixelBuffer, Rectangle2D> callback = pixelBuffer -> {
+     *     ByteBuffer buf = ((PixelBuffer<ByteBuffer>) pixelBuffer).getBuffer();
+     *     // Update the buffer.
+     *     return new Rectangle2D(x, y, width, height);
+     * };
+     * pixelBuffer.updateBuffer(callback);
+     * }</pre>
+     *
+     * @param callback the {@link javafx.util.Callback} method which updates the {@code java.nio.Buffer}.
+     * @throws IllegalStateException if this method is not being called on the JavaFX Application Thread.
+     * @throws NullPointerException  if {@code callback} is {@code null}.
```

Comment:
As with `@param` and `@return` we don't typically end `@throws` statements with a period